src/Lykke.Job.HistoryExportBuilder.Services/CsvMaker.cs

```diff
@@ -0,0 +1,112 @@
+
+        public CsvMaker(
+            IAssetsServiceWithCache assetsServiceWithCache)
+        {
+            _assetsServiceWithCache = assetsServiceWithCache;
+        }
+        
+        public async Task<MemoryStream> MakeAsync(IEnumerable<HistoryOperation> operations)
+        {
+            var assets = await _assetsServiceWithCache.GetAllAssetsAsync();
+            var assetPairs = await _assetsServiceWithCache.GetAllAssetPairsAsync();
+            
+            using (var stream = new MemoryStream())
+            {
+                using (var streamWriter = new StreamWriter(stream) {AutoFlush = true})
+                {
+                    var userCsv = new CsvWriter(streamWriter);
+                    userCsv.Configuration.RegisterClassMap<HistoryOperationCsvEntryMap>();
+
+                    userCsv.WriteRecords(operations.Select(x =>
+                    {
+                        var assetPair = assetPairs.FirstOrDefault(y => y.Id == x.AssetPair);
+                        
+                        decimal baseAmount;
+                        decimal? quoteAmount;
+
+                        if (assetPair == null)
+                        {
+                            baseAmount = Convert.ToDecimal(x.Amount);
+                            quoteAmount = null;
+                        }
+                        else
+                        {
+                            baseAmount = assetPair.BaseAssetId == x.Asset
+                                ? Convert.ToDecimal(x.Amount)
+                                : Convert.ToDecimal(x.Amount) * Convert.ToDecimal(x.Price);
+                            quoteAmount = assetPair.BaseAssetId == x.Asset
+                                ? Convert.ToDecimal(x.Amount) * Convert.ToDecimal(x.Price)
+                                : Convert.ToDecimal(x.Amount);
+                        }
+                        
+                        return new HistoryOperationCsvEntry
+                        {
+                            Date = x.DateTime,
+                            Type = x.Type.ToString(),
+                            Exchange = "Lykke",
+                            BaseAmount = baseAmount,
+                            BaseCurrency = assets.Single(y => y.Id == (assetPair != null ? assetPair.BaseAssetId : x.Asset)).DisplayId,
+                            QuoteAmount = quoteAmount,
+                            QuoteCurrency = assetPair?.QuotingAssetId != null ? assets.Single(y => y.Id == assetPair.QuotingAssetId).DisplayId : null,
+                            FeeCurrency = assets.Single(y => y.Id == x.Asset).DisplayId,
+                            Fee = Convert.ToDecimal(x.FeeSize)
+                        };
+                    }));
+                        
+                    stream.Seek(0, SeekOrigin.Begin);
+                            
+                    stream.Position = 0;
+
+                    return stream;
```

Comment:
Disposed memory stream will be returned - consider disposing it after usage from calling code